### PR TITLE
Iteration4

### PR DESCRIPTION
**Pull request by srslie**
**Changes:**
- Add tests for Activity class
- Add Activity class to represent activity data objects (each is one day's log for each user)
- Add methods to Activity to convert steps to miles and verify if step goal is met
- Add tests for CommunityActivity class
- Add CommunityActivity class that holds all Activity objects
- Adds methods to find activities by date and grab averages for all users, specific-to-user stats like all-time stair climbing record or days they reached step goals
- Add basic displays in script with minor additions to HTML

**Issues:**
- I'm still hoping to get feedback on the tests: I wasn't sure how to get around having a somewhat lengthy data set for the CommunityActivities, since I wanted to test multiple instances and we had some calculations to verify a week's worth of data.
- It looks like there was some discrepancy between the methods they asked to include and the ones they asked to display. Let's touch base about which ones are important to us to display.
- There's a lot of data here and I haven't touched CSS, so I'd love input as to how to display all the stuff in activities and if we need to make more divs or what. Your eye for design is appreciated!
- Let me know if anything looks like it needs a refactor. There's some room for DRYing it up I'm sure, like in the loops that display the graphs, but since we might be changing that with chart.js, I just wanted to put that up. I'm sure the code could do with a good once-over though if you see anything.
- Should our Community classes be plural? CommunityActivities vs CommunityActivity?

- [x] DRY and SRP (ish)
- [x] Tested Locally

**How to Review Changes or Manually Test Code:**
- Open Activity.js and Activity-test.js to view class and corresponding test to pass, uncomment out the requires and module.exports at the top of each file
- Open CommunityActivity.js and CommunityActivity-test.js to view class and corresponding test to pass,  uncomment out the requires and module.exports at the top of each file
- Open scripts.js to view code and DOM manipulation, re-comment out the requires and module.exports in all files
- Open index.html in browser to view displays 

**Notes:**
- I did a lot of this pretty late at night so apologies if anything's sloppy. Let me know any changes and I'll be happy to change it!
- My code editor Atom is bugging out on my computer and I had to switch to VSCode, which I'm still learning. I think it somewhat disrupted my normal commit cycle just because I'm used to working with Atom sidebars for git, and so I had fewer commits than I would've liked. I will strive to be better about them going forward!

### CODE REVIEW

File: tests/CommunityActivity-test.js [hunk 1]
```diff
@@ -0,0 +1,398 @@
...
+      "friends": [
+        9,
+        27,
+        21,
+        13
+      ]
+    });
+
+  });
+
```
Comment (conconartist) [inline tests/CommunityActivity-test.js:327]: NIce job including a variety of data!

File: tests/CommunityActivity-test.js [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+      "date": "2019/06/17",
+      "numSteps": 13750,
+      "minutesActive": 65,
+      "flightsOfStairs": 4
+    })
+
+    expect(communityActivity.daysExceedingStepGoal(user)[0]).to.deep.equal(activity3)
+    expect(communityActivity.daysExceedingStepGoal(user2).length).to.equal(9)
+  });
+
```
Comment (conconartist) [inline tests/CommunityActivity-test.js:373]: Maybe a path to verify if they have not exceeded their step goal?

File: tests/CommunityActivity-test.js [hunk 3]
```diff
@@ -0,0 +1,398 @@
...
+    expect(communityActivity2.findCommunityAverage("2019/06/15", 'numSteps')).to.equal(6027)
+    expect(communityActivity3.findCommunityAverage("2019/09/02", 'numSteps')).to.equal(7869)
+  });
+
+  it('should return community average minutes active on a specific date', function() {
+    expect(communityActivity2.findCommunityAverage("2019/06/15", 'minutesActive')).to.equal(144)
+    expect(communityActivity3.findCommunityAverage("2019/09/02", 'minutesActive')).to.equal(219)
+  });
+
+});
```
Comment (conconartist) [inline tests/CommunityActivity-test.js:419]: Do we want to add test paths for weekly averages?  'should find average minutes active over a given week' for user 50 using their start/end date?

File: src/scripts.js [hunk 4]
```diff
@@ -75,6 +85,103 @@ const showHydrationStatsWeek = (startDate, endDate) => {
...
+
+  let activitiesDisplay = weekActivities.reduce((display, activity) => {
+    display += `<p>${activity.date}: ${activity.stairsClimbed} stairs climbed</p>`
+    return display
+  }, '') 
+
+  graphTitle.insertAdjacentHTML('afterend',
+    `<article class="stairs-stats-week">Stairs Climbed over the week of ${startDate}-${endDate}:` + activitiesDisplay + '</article>')
+}
+
```
Comment (conconartist) [inline src/scripts.js:167]: Your code looks really DRY to me-- I would like to see if/how we can combine some of our functions to work across categories when all 4 iterations are done.

**Review by conconartist — CHANGES_REQUESTED** [review]
I think it looks great overall- thanks for your work on this! I added a few comments regarding the testing and suggestions for future iterations. I agree on some of the spec being unclear about what data to display-- I think what you have looks great for now, to just get it all displayed first and then we can decide what we want to toggle and how we want to incorporate the graph views when we get there. As far as testing, I'd lean more on the side of having more data on the tests than too little, but that's just my opinion, so I'd at least have a week's worth (if not 2) of data to test the weekly stats.

File: tests/CommunityActivity-test.js (hunk 2 above)
Comment (srslie) [reply tests/CommunityActivity-test.js:373]: Added!

File: src/scripts.js (hunk 4 above)
Comment (srslie) [reply src/scripts.js:167]: Thanks and I agree, I think we can might be able to grab out the functions that get used in the multiple classes somehow--maybe see if we use something like "intake a userID and return the instance of the class on this date or for this date range or for all time" and make that into pure functions that we can then use to get variables we pass into the methods or something? We can probably figure out some way to make it DRYer.

**Comment by srslie** [comment]
I made those changes so if this looks good, feel free to merge! Otherwise let me know what it needs :)

**Pull request merged**